### PR TITLE
Widgets: Script Access Robustness

### PR DESCRIPTION
**Pull request by runrevmark**
This branch contains a number of changes to improve the robustness of using widgets which interact with LiveCode Script.

Syntax for converting an int to and from a codepoint has been added - `the code of <char>` and `the char with <code>`.

Syntax for dispatch has been removed and replaced with send (just the verb has changed).

A variant of 'dispatch' - 'post' has been added. This is the same as send except that the message is queued for delivery at the next wait point, and you can't get a return value (as it executes after the statement finishes executing!).

Syntax for 'the local time' is now 'the local date'.

New syntax for 'the universal time' (returning the number of seconds inc fraction, since the start of the UNIX Epoch) has been added.

The OnSave / OnLoad / OnCreate / OnDestroy / OnPaint no longer allow syntax to be executed which require script access.

All the existing widgets have been updated to use the new syntax and not call any script-access requiring functions beyond post.

### CODE REVIEW

**Comment by runrevmark** [comment]
If someone could check the date module for portability - particularly to Win32, that would be great!

File: engine/src/widget.cpp [hunk 1]
```diff
@@ -956,12 +975,16 @@ void MCWidget::OnHitTest(const MCRectangle& p_intersect, bool& r_hit)
 {
     fprintf(stderr, "MCWidget::OnHitTest\n");
     r_hit = maskrect(p_intersect);
+    
+    // In theory this handler shouldn't allow script access.
```
Comment (livecodefraser) [inline engine/src/widget.cpp:979]: i.e we've forgotten to implement these events...

(and, by "we", I of course mean "I")

File: libscript/src/module-date.cpp [hunk 2]
```diff
@@ -19,6 +19,7 @@ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
 #include <foundation-auto.h>
 
 #include <time.h>
+#include <sys/time.h>
```
Comment (livecodefraser) [inline libscript/src/module-date.cpp]: Makes MSVC sad :-(

**Comment by livecodefraser** [comment]
Reviewed-By: Fraser Gordon

**Pull request merged**